nova/virt/vmwareapi/session.py

```diff
@@ -136,6 +136,10 @@ def _call_method(self, module, method, *args, **kwargs):
                     # We have found the argument with the moref
                     # causing the exception and we can try to recover it
                     arg.fetch_moref(self)
+                    if not arg.moref:
+                        # We didn't recover the reference
+                        ctxt.reraise = True
+                        break
```

Comment:
Can this trigger for volumes with `arg.fetch_moref()` raising an exception? Does it trigger for VMs?

Reply:
Isn't `ctxt.rereaise = True` the default? We just put it here explicitly as a reminder?

Reply:
The idea was that the `arg.fetch_moref()` should raise the specific exception for the resource, so we wouldn't reach that block. In the case of volumes, no exception was risen, so we ended up with the attribute error.

Reply:
> Isn't ctxt.rereaise = True the default? We just put it here explicitly as a reminder?

Well, it is a loop, and a couple of lines below it gets set to `False`. While it should not happen that it gets set to `False` below and then end up in the break here is from my point of view not immediately evident. Setting it explicitly `True´ plus `break` the context one needs to understand what is happening is limited to those two lines.

Reply:
will there be case that moref is None at line 129 and we never land in for loop, cause arg.moref to be not updated before going for _call_method at line 152 again ?

Reply:
Passing a reference with `None` to  `invoke_api` is invalid and will raise a different exception then the one we try to handle here.

So, no reference passed to `_call_method` can be `None` prior our call to `arg.fetch_moref`.
I could skip on the loop right away, but I think, that is out of scope of this change

If `moref is None` in line 129, because the api didn't tell us which reference could not be resolved, we will simply iterate over all arguments and none will match `None`.